### DISCUSSION TITLE
Unable to clone repository using ssh

### DISCUSSION THREAD

**Discussion by Hoet** [community/community#46057] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

using windows 10
Putty as my regular ssh client with Pageant as ssh-agent

When I try to clone a test repository, I get a Permission denied. See below for the log.
git clone git@github.com:Hoet/mauricetest.git
Cloning into 'mauricetest'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

When I connec via the putty client to port 22 I get the following output

Using username "git".
Authenticating with public key "" from agent
Server refused to allocate pty
Hi Hoet! You've successfully authenticated, but GitHub does not provide shell access.

So it seems that my key is correct and accepted.

Never had any issues working this way. Normaly I connect to GitLab repositories. It has been ages the last time I connected to GitHub.

**Comment by norwd** [comment]
Hi @Hoet,

It sounds like you might need to add your SSH key to your GitHub account (you can do this by going to https://github.com/settings/keys). After you've done that, (assuming you haven't already), have a read through [this](https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey) page that should walk you through the troubleshooting steps.

> Error: Permission denied (publickey)
A "Permission denied" error means that the server rejected your connection. There could be several reasons why, and the most common examples are explained below.

**Reply by Hoet** [reply]
Thanks for the answer. I can connec to git successfully with my putty ssh client (with the right key orcourse). But when I try to make the connection via git it fails. It seems like that git is not able to detect the key that is used by Putty. While the same set-up is working just fine for agitlab repository. 

My pub key was added to the setting -> ssh & GPG keys. And it is working. Else putty would not hae been able to make a connection.

**Reply by norwd** [reply]
Try `ssh -vT git@github.com` to get a [detailed output](https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey#getting-more-details)?

**Reply by Hoet** [reply]
I do not have a ssh commandline tool on my windows. I have never needed a ssh command on my windows system. just git and the right key in putty's ssh-agent. 

I've not tried cygwin, that would add a ssh command to windoze... never needed that before for git connections over ssh

**Reply by norwd** [reply]
> I do not have a ssh commandline tool on my windows.

According to [this](https://learn.microsoft.com/en-us/windows/terminal/tutorials/ssh#access-windows-ssh-client) article, you should already have OpenSSH installed in Windows.

> By default, the OpenSSH client will be located in the directory: C:\Windows\System32\OpenSSH. You can also check that it is installed in Windows Settings > Apps > Optional features, then search for "OpenSSH" in your installed features.

If you run `ssh -vT git@github.com` in your terminal of choice does it complain about not having `ssh` in your path? If so you can add `C:\Windows\System32\OpenSSH` to your `PATH` variable and it will become available 🙂

In any case, if you are having trouble cloning from GitHub using ssh for any reason, might I suggest using the https://desktop.github.com client? It authenticates to GitHub directly using your username and password in the browser and after that one time setup you never need to worry about authentication again. It has a good GUI but if you prefer to use a terminal, the `git` command will piggyback off GitHub Desktop's authentication. I highly recommend it 👍

**Comment by gaganachiranjitha99** [comment]
cmt1

**Comment by bondarenkod** [comment]
@Hoet  try this: https://gist.github.com/bondarenkod/20e96267a46f78bf98d3e999d04c102f

**Reply by vitalykarasik** [reply]
Many thanks, I spent 2+ hours on this stupid issue, you made my day! :-)

**Comment by Hasher4856** [comment]
It sounds like you're on the right track—especially since your key is recognized and you’re seeing the GitHub "no shell access" message. From what you’ve described, it’s likely a mismatch between the public key you're using and what GitHub has on file. GitHub is strict about exact key pairing, especially when switching between platforms like GitLab and GitHub.

A quick fix might be to re-add your public key from Pageant to GitHub via [your SSH settings](https://github.com/settings/keys). Also, double-check you're using the correct private key format for PuTTY (PPK vs PEM) and that Pageant is running with the correct loaded key. GitHub may not recognize your PuTTY key unless it’s converted properly.

This reminds me of how compatibility and precision matter in other fields too. For example, on [Cartea](https://www.icartea.com/), we review how small mismatches in car tech systems—like diagnostics software or firmware—can lead to frustrating issues. Just like this, it's usually not a big flaw—just a small detail that needs syncing up!

Hope this helps you get back to smooth cloning.

Let me know if you'd like a shorter version or more technical details!










Humanize 189 words

**Comment by businesslancing56-boop** [comment]
I also recently faced a similar SSH cloning issue while working on a private Git repository used to manage frontend assets for a custom Shopify storefront I'm building for the fashion brand I manage, [Ariva Streetwear](https://arivastreetwear.com/).

The issue in our case was due to a missing SSH key entry in the local .ssh/config combined with a misconfigured Git remote URL format. Switching from 'git@github.com: user/repo. git' to the full SSH format and ensuring the key had proper read/write access resolved the issue.

For anyone still stuck, I recommend:

Double-checking your SSH agent is running (ssh-add -l)

Making sure Git is using the correct identity (GIT_SSH_COMMAND)

Verifying repository access with ssh -T git@github.com

Hope this helps someone else facing a similar problem.

Thanks.

**Comment by corazulshop-netizen** [comment]
Just like connecting with experienced alumni can guide students toward success, working with seasoned professionals is essential for achieving outstanding results in any field. In construction and design, choosing a team specialized in [Rénovation haut de gamme Laval](https://groupecasarenova.ca/) ensures you benefit from proven expertise, personalized guidance, and high-quality execution from start to finish."

**Comment by asma019** [comment]
Hi @Hoet,
Your SSH key is working fine (as shown by the “successfully authenticated” message), but GitHub is refusing the git clone because Git is not picking up the same key configuration that PuTTY/Pageant is using.

A few things you can try:

Check the remote URL format – Make sure it matches the GitHub SSH syntax exactly:

`git clone git@github.com:Hoet/mauricetest.git`


(no typos, and repository exists under your account).

Ensure Git for Windows is using PuTTY’s agent – By default, Git for Windows uses OpenSSH. If you’re using PuTTY/Pageant, set:

`set GIT_SSH="C:\Program Files\PuTTY\plink.exe"`


or configure it in your Git config so Git commands go through PuTTY’s SSH client.

Test with OpenSSH instead of PuTTY – If possible, try adding your key to OpenSSH’s ssh-agent and running:

`ssh -T git@github.com`


This confirms whether it’s a PuTTY integration issue.

Verify key is added to GitHub – Go to [GitHub → Settings → SSH and GPG keys](https://github.com/settings/keys) and confirm your public key is present (it might be different from the one you use for GitLab).

Since you haven’t connected to GitHub in a while, your key there may be outdated or missing. Adding your current public key should solve it.
